lib/grammers-client/src/types/message_update.rs

```diff
@@ -0,0 +1,42 @@
+/// Occurs whenever a message is deleted.
+/// Note that this event isn’t 100% reliable, since Telegram doesn’t always notify the clients that a message was deleted.
+///
+/// When `MessageDeletion#channel_id` is Some, it means the message was deleted from a channel.
+#[derive(Debug)]
+pub struct MessageDeletion {
+    pub(crate) channel_id: Option<i64>,
+    pub(crate) messages: Vec<i32>,
+}
+
+impl MessageDeletion {
+    /// Creates a new `MessageDeletion` from a vector of message IDs that is deleted in a channel.
+    pub(crate) fn new_with_channel(messages: Vec<i32>, channel: i64) -> Self {
+        Self {
+            channel_id: Some(channel),
+            messages,
+        }
+    }
+
+    /// Creates a new `MessageDeletion` from a vector of message IDs.
+    pub(crate) fn new(messages: Vec<i32>) -> Self {
+        Self {
+            channel_id: None,
+            messages,
+        }
+    }
+
+    /// Returns the channel ID if the message was deleted from a channel.
+    pub fn channel_id(&self) -> Option<i64> {
```

Comment:
I think this should do something similar to `Message::chat` (and maybe rename this to `chat` too).

https://github.com/Lonami/grammers/blob/master/lib/grammers-client/src/types/message.rs#L234-L240

I would like to hide away the raw API as much as possible where it makes sense. Users do not need to know only the channel ID is present (it's confusing, as some groups are channels too). Renaming and updating the documentation to say "this information is only known for megagroups and broadcast channels, and only the `id` is present" would be a better API IMO.

Reply:
Does this mean I should include a client inside the `MessageDeletion` or should that method accept a client as parameter? Since in my understanding this involves fetching data about the chat with the given `channel_id`?